plugins/scaffolder-backend/src/scaffolder/storage/disk.ts

```diff
@@ -59,43 +63,38 @@ export class DiskStorage implements Base {
   }
 
   private async index(): Promise<DiskIndexEntry[]> {
-    return new Promise((resolve, reject) => {
-      glob(`${this.repoDir}/**/template-info.json`, async (err, matches) => {
-        if (err) {
-          reject(err);
-        }
+    const matches = await globby(`${this.repoDir}/**/template-info.json`);
 
-        const fileContents: Array<{
-          location: string;
-          contents: string;
-        }> = await Promise.all(
-          matches.map(async (location: string) => ({
-            location,
-            contents: await fs.readFile(location, 'utf-8'),
-          })),
-        );
+    const fileContents: Array<{
+      location: string;
+      contents: string;
+    }> = await Promise.all(
+      matches.map(async (location: string) => ({
+        location,
+        contents: await fs.readFile(location, 'utf-8'),
+      })),
+    );
 
-        const validFiles = fileContents.reduce(
-          (diskIndexEntries: DiskIndexEntry[], currentFile) => {
-            try {
-              const parsed: Template = JSON.parse(currentFile.contents);
-              return [
-                ...diskIndexEntries,
-                { location: currentFile.location, contents: parsed },
-              ];
-            } catch (ex) {
-              logger.error('Failure parsing JSON for template', {
-                path: currentFile.location,
-              });
-            }
+    const validFiles = fileContents.reduce(
+      (diskIndexEntries: DiskIndexEntry[], currentFile) => {
+        try {
+          const parsed: Template = JSON.parse(currentFile.contents);
+          return [
+            ...diskIndexEntries,
```

Comment:
this one copies and recreates the entire array every round of the loop - at least this one feels like something that would have been better and clearer as a for-of+push :)

Reply:
+1 for regular for-loop